### ISSUE TITLE
Passive cancel of observe

### ISSUE THREAD

**Issue by Redferne** [chrysn/aiocoap#134]
Using aiocoap with LWM2M client from https://github.com/aellwein/lwm2mclient

I ran into an issue with Leshan LWM2M server which has only implemented passive cancellation of observations. https://tools.ietf.org/html/rfc7641#section-3.6

Here's the client sending a periodic observe notification. The observe has been cancelled on the server and thus is reponds with RST and empty token. How would it be possible to relay this to upper layers so that the serverobservation objects get aware and can kill any notify producers?

```
2018-12-18 19:28:49,732 [DEBUG] Sending message <aiocoap.Message at 0x7f096ed4aef0: Type.CON 2.05 Content (MID 50769, token dd4066121212f5f5) remote <aiocoap.transports.tinydtls.DTLSClientConnection object at 0x7f096eda6278>, 2 option(s), 11 byte(s) payload>
2018-12-18 19:28:49,732 [DEBUG] Exchange added, message ID: 50769.
2018-12-18 19:28:49,786 [DEBUG] Incoming message <aiocoap.Message at 0x7f096ed45cf8: Type.RST EMPTY (MID 50769, empty token) remote <aiocoap.transports.tinydtls.DTLSClientConnection object at 0x7f096eda6278>>
2018-12-18 19:28:49,786 [DEBUG] New unique message received
```

**Comment by chrysn** [comment]
just to make sure I'm getting the roles right (given that LWM2M has slightly unusual choices here): The aiocoap component is sending notifications to a leshan device that has requested them, but is no longer interested; the Leshan device sends RST but aiocoap still sends new messages?

**Comment by Redferne** [comment]
Yes. Absolutely correct. I've implemented the *update_observation_count* and it increases for each observation request, which is as expected. Since Leshan server does not send any active cancel but only responds with RST when observation is cancelled serverside, newcount never decreases. Also tried to implement *add_observation* and set the accept() callback. But same goes here, the callback is never called. The RST response to the CON notification is removed from the exchange list in the dispatch_message function. I've tried to come up with a shortcut from the context/message layer (mid mapping) directly to the resource layer for triggering the observe cancel, but I'm no seasoned Python developer :sob:
